### PR TITLE
E2E automation and other test changes

### PR DESCRIPTION
**Pull request by shrishrirang**
- Revamped the e2e test directory structure. e2etest\shared directory contains files common to all platforms (Browser and Cordova) with the exception of the Platform directory.
- "npm run testsetup" copies the SDK files to the appropriate location for the browser and Cordova test apps to use. No more static SDK references checked into the repository.
- Browser test project is no more coupled with WinJS. You can run browser tests by running _http-server_ from e2etest\browser or by opening e2etest\browser\browserEndToEndTests.sln in VS.
- e2etest\scripts\runTests.js automates the e2e tests for Cordova. Currently the automation has hardcoded references to Android as that is the only platform automation runs on. Extending it to iOS should be straightforward.

### CODE REVIEW

**Comment by shrishrirang** [comment]
\+ @danderson00

File: e2etest/shared/platform/cordova/platformSpecificFunctions.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ----------------------------------------------------------------------------
+
+function createPlatformSpecificFunctions() {
+
+    // Currently only implemented for Android
+    function getApplicationParam(name) {
+        var config = {};
+
+        var def = $.Deferred();
```
Comment (danderson00) [inline e2etest/shared/platform/cordova/platformSpecificFunctions.js]: Do we have an explicit dependency on jQuery?

Comment (shrishrirang) [inline e2etest/shared/platform/cordova/platformSpecificFunctions.js]: Yes. e2etests have a dependency on jQuery. Just to re-state the obvious - The SDK does not depend on jQuery.

Comment (danderson00) [inline e2etest/shared/platform/cordova/platformSpecificFunctions.js]: Aren't platformSpecificFunctions consumed in the build though?

**Comment by shrishrirang** [comment]
Ping! :)

**Comment by danderson00** [comment]
Looks good

**Pull request merged**